From 5053589994b3e92bb4f27d1e733082760395eb49 Mon Sep 17 00:00:00 2001
From: Philip-Dylan Gleonec <philip-dylan.gleonec@savoirfairelinux.com>
Date: Tue, 13 Apr 2021 16:34:42 +0200
Subject: [PATCH] openvswitch.service: require loaded iommu

Add a ConditionDirectoryNotEmpty towards /sys/class/iommu to ensure
the openvswitch services do not start until the iommu is properly
loaded.
This is required as openvswitch without iommu makes dpdk use the
`uio_pci_generic` driver, which requires physical adresses and require
broad filesystem access than the `vfio_pci` driver.
Thus, to increase security, force theuse of `vfio_pci`.

Signed-off-by: Philip-Dylan Gleonec <philip-dylan.gleonec@savoirfairelinux.com>
---
 rhel/usr_lib_systemd_system_openvswitch.service | 1 +
 1 file changed, 1 insertion(+)

diff --git a/rhel/usr_lib_systemd_system_openvswitch.service b/rhel/usr_lib_systemd_system_openvswitch.service
index feaba37d5..46b3b21f6 100644
--- a/rhel/usr_lib_systemd_system_openvswitch.service
+++ b/rhel/usr_lib_systemd_system_openvswitch.service
@@ -5,6 +5,7 @@ After=network-pre.target ovsdb-server.service ovs-vswitchd.service
 PartOf=network.target
 Requires=ovsdb-server.service
 Requires=ovs-vswitchd.service
+ConditionDirectoryNotEmpty=/sys/class/iommu
 
 [Service]
 Type=oneshot
-- 
2.25.1

